### DISCUSSION TITLE
Tree filter state lost when selecting node

### DISCUSSION THREAD

**Discussion by joneves** [dowjones/react-dropdown-tree-select#446] [Q&A]
Hi

Just wondering if this is the desired logic or if my implementation isn't working correctly.  Happy to raise a bug report if this is found to be an issue.

Looks as if on every prop change the TreeManager in initialized from scratch therefore filter state is lost.   

Thanks

**Comment by mrchief** [comment]
Is there an event handler that changes your global state which then re-renders every child?

**Comment by joneves** [comment]
Thanks for replying, yes there is, and as i discovered this morning this is causing the re render and tree initialize.  But I need to update an external store and run validation, so the parent will need to re render.  I was trying to use the component in a controlled way, but that doesn't seem possible, I'm now unsure as to how i get the value when the tree closes.  How do developers normally handle this?

**Comment by mrchief** [comment]
There's prevent tree render on parent render HOC example in the docs that you can refer to. 

Normally you'd hook up `onChange` or `onNodeToggle` to a handler which then updates your global state (redux, hooks, mobx or anything that manages a page or app state). You can even go as far as saving/updating the initial `data` that the component receives in those handlers so even if the component is reinitialized, it receives the most up-to-date state.

**Comment by xmakina** [comment]
Working example in React Hooks https://github.com/dowjones/react-dropdown-tree-select/issues/445#issuecomment-734005801
